tsconfig.json

```diff
@@ -0,0 +1,9 @@
+{
```

Comment:
@liamsorsby could you talk a little about the intent behind this tsconfig.json file?  Is this just here to allow someone to use `tsc` to generate the `types.d.ts` file, or are there more reasons for having this here?

Reply:
@astorm Hey, yes this was just created to ensure that the config used to generate the types file was consistent. However, now I think about it, this should really be a step in the CI and pushed as part of the build tagging.